### PR TITLE
[MM-34457] Add E2E webapp tests to CI

### PR DESCRIPTION
**Pull request by hanzei**
#### Summary
To ensure the go E2E webapp tests are validated on every change, this PR adds them to CircleCI. The changes are mostly based on work of the q

The PR is based on https://github.com/mattermost/mattermost-plugin-apps/pull/115.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-34457
https://mattermost.atlassian.net/browse/MM-34761
https://mattermost.atlassian.net/browse/MM-34780

### CODE REVIEW

**Review by levb — COMMENTED** [review]
Is there a way to point this test at a specific branch of `-webapp`? Like, if I make a breaking change in the plugin, and have a matching webapp branch, how do I use it for the test?

**Comment by hanzei** [comment]
@levb Great question. In https://github.com/mattermost/mattermost-plugin-apps/pull/124/files#diff-78a8a19706dbd2a4425dd72bdab0502ed7a2cef16365ab7030a5a0588927bf47R109 the Docker image tag is set. On a breaking change a dev would update that tag to point to the corresponding webapp PR. AFAIK there is not easy mechanism to point to a branch besides `master`. Do we need that?

**Comment by mickmister** [comment]
@hanzei CI tests are currently failing due to the webapp being built from the docker tag in the circleci config, as explained above. The webapp branch `feature/cloud-apps` needs to be used instead. Do we need to add steps to build the webapp from source?

**Comment by hanzei** [comment]
@mickmister The commit that is being used here is https://github.com/mattermost/mattermost-webapp/commit/77d8b88e09be9df8693d8397845b9af43dc6fb38, which is part of `feature/cloud-apps-rebase`. Is that correcet?

File: server/examples/go/hello/http_hello/functions.go [hunk 1]
```diff
@@ -21,6 +22,7 @@ func (h *helloapp) GetBindings(w http.ResponseWriter, req *http.Request, claims
 func (h *helloapp) Install(w http.ResponseWriter, req *http.Request, claims *apps.JWTClaims, c *apps.CallRequest) (int, error) {
 	out, err := h.HelloApp.Install(AppID, AppDisplayName, c)
 	if err != nil {
+		log.Printf("err: %#+v\n", err)
```
Comment (larkox) [inline server/examples/go/hello/http_hello/functions.go:25]: Is this intended? If so, should it be more verbose? ("Error installing Hello App" or something like this)

File: server/examples/go/hello/install.go [hunk 2]
```diff
@@ -14,6 +15,8 @@ import (
 
 func (h *HelloApp) Install(appID apps.AppID, channelDisplayName string, c *apps.CallRequest) (md.MD, error) {
 	bot := mmclient.AsBot(c.Context)
+	log.Printf("c.Context: %#+v\n", c.Context)
```
Comment (larkox) [inline server/examples/go/hello/install.go:18]: Are these intended?

File: server/proxy/install.go [hunk 3]
```diff
@@ -95,6 +96,7 @@ func (p *Proxy) InstallApp(sessionID, actingUserID string, cc *apps.Context, tru
 	}
 	resp := p.Call(sessionID, actingUserID, creq)
 	if resp.Type == apps.CallResponseTypeError {
+		log.Printf("esp: %#+v\n", *resp.CallResponse)
```
Comment (larkox) [inline server/proxy/install.go:99]: Is this intended? If so, shouldn't be it `resp` instead of `esp`?

**Review by larkox — APPROVED** [review]

**Comment by hanzei** [comment]
Sorry @larkox, forgot to withdraw the request for review. This is WIP.

**Review by levb — APPROVED** [review]

**Pull request merged**